data/scripts.js

```diff
@@ -159,154 +167,150 @@ exports.BattleScripts = {
+						if (!pokemon.hp) {
+							this.faint(pokemon, pokemon, move);
+						}
 
-		let damage = false;
-		if (move.target === 'all' || move.target === 'foeSide' || move.target === 'allySide' || move.target === 'allyTeam') {
-			damage = this.tryMoveHit(target, pokemon, move);
-			if (damage || damage === 0 || damage === undefined) moveResult = true;
-		} else if (move.target === 'allAdjacent' || move.target === 'allAdjacentFoes') {
-			if (!targets.length) {
-				this.attrLastMove('[notarget]');
-				this.add('-notarget');
-				return false;
-			}
-			if (targets.length > 1) move.spreadHit = true;
-			let hitTargets = [];
-			for (let i = 0; i < targets.length; i++) {
-				let hitResult = this.tryMoveHit(targets[i], pokemon, move);
-				if (hitResult || hitResult === 0 || hitResult === undefined) {
-					moveResult = true;
-					hitTargets.push(targets[i].toString().substr(0, 3));
-				}
-				if (damage !== false) {
-					damage += hitResult || 0;
-				} else {
-					damage = hitResult;
-				}
-			}
-			if (move.spreadHit) this.attrLastMove('[spread] ' + hitTargets.join(','));
-		} else {
-			target = targets[0];
-			let lacksTarget = target.fainted;
-			if (!lacksTarget) {
-				if (move.target === 'adjacentFoe' || move.target === 'adjacentAlly' || move.target === 'normal' || move.target === 'randomNormal') {
-					lacksTarget = !this.isAdjacent(target, pokemon);
+						if (!moveResult) {
+							this.singleEvent('MoveFail', move, null, target, pokemon, move);
+						} else {
+							if (!move.negateSecondary && !(move.hasSheerForce && pokemon.hasAbility('sheerforce'))) {
+								this.singleEvent('AfterMoveSecondarySelf', move, null, pokemon, target, move);
+								this.runEvent('AfterMoveSecondarySelf', pokemon, target, move);
+							}
+							if (pokemon.moveThisTurnSucceeded === oldMoveSucceeded) pokemon.moveThisTurnSucceeded = true;
+							return true;
+						}
+					}
 				}
 			}
-			if (lacksTarget && (!move.flags['charge'] || pokemon.volatiles['twoturnmove'])) {
+			if (noTarget) {
 				this.attrLastMove('[notarget]');
 				this.add('-notarget');
 				if (move.target === 'normal') pokemon.isStaleCon = 0;
-				return false;
 			}
-			damage = this.tryMoveHit(target, pokemon, move);
-			if (damage || damage === 0 || damage === undefined) moveResult = true;
-		}
-		if (move.selfBoost && moveResult) this.moveHit(pokemon, pokemon, move, move.selfBoost, false, true);
-		if (!pokemon.hp) {
-			this.faint(pokemon, pokemon, move);
-		}
-
-		if (!moveResult) {
-			this.singleEvent('MoveFail', move, null, target, pokemon, move);
-			return false;
-		}
-
-		if (!move.negateSecondary && !(move.hasSheerForce && pokemon.hasAbility('sheerforce'))) {
-			this.singleEvent('AfterMoveSecondarySelf', move, null, pokemon, target, move);
-			this.runEvent('AfterMoveSecondarySelf', pokemon, target, move);
 		}
-		return true;
+		if (pokemon.moveThisTurnSucceeded === oldMoveSucceeded) pokemon.moveThisTurnSucceeded = false;
```

Comment:
What's this doing?

Reply:
This lets the result of called moves override the result of the caller; it’s necessary to do something like this due to the recursive nature of the battle engine wherein called moves are resolved before the move that called them.